### PR TITLE
fix(workflow): Convert issueList actions to React.Component

### PR DESCRIPTION
**Pull request by taylangocmen**
This makes issueList a React.Component instead of using createReactClass. More prep work for inbox.

### CODE REVIEW

File: src/sentry/static/sentry/app/types/index.tsx [hunk 1]
```diff
@@ -187,7 +187,7 @@ export type AvatarProject = {
 /**
  * Simple timeseries data used in groups, projects and release health.
  */
-export type TimeseriesValue = [timestamp: number, value: number];
```
Comment (billyvg) [inline src/sentry/static/sentry/app/types/index.tsx:190]: Why remove this?

File: src/sentry/static/sentry/app/views/issueList/actions.tsx [hunk 2]
```diff
@@ -176,36 +167,21 @@ type State = {
...
-    groupIds: PropTypes.instanceOf(Array).isRequired,
-    onRealtimeChange: PropTypes.func.isRequired,
-    onSelectStatsPeriod: PropTypes.func.isRequired,
-    realtimeActive: PropTypes.bool.isRequired,
-    statsPeriod: PropTypes.string.isRequired,
-    query: PropTypes.string.isRequired,
-    queryCount: PropTypes.number,
-  },
-
-  mixins: [Reflux.listenTo(SelectedGroupStore, 'handleSelectedGroupChange') as any],
```
Comment (billyvg) [inline src/sentry/static/sentry/app/views/issueList/actions.tsx:197]: Just a note, you'll need to initialize component state with the store state, on mount

File: src/sentry/static/sentry/app/types/index.tsx (hunk 1 above)
Comment (taylangocmen) [reply src/sentry/static/sentry/app/types/index.tsx:190]: My IDE was giving me a warning but I realize that this is valied TS. I'll put it back

Comment (billyvg) [reply src/sentry/static/sentry/app/types/index.tsx:190]: your TS version is probably outdated in your IDE

**Review by scttcper — APPROVED** [review]

**Pull request merged**